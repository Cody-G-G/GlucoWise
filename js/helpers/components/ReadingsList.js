'use strict';
import React, {Component} from 'react';
import {ListView, Text, TouchableOpacity, View} from 'react-native';
import {Icon} from 'native-base';
import TextBold from "./TextBold";
import styles from "./styles";
import dateUtil from "../util/date";
import processReading from "../util/readingProcessor";

export default class ReadingsList extends Component {
    constructor(props) {
        super(props);
        this.ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
    }

    render() {
        return (
            <ListView dataSource={this.ds.cloneWithRows(this.props.readings)} enableEmptySections={true} renderRow={(rowData) =>
                    <View style={styles.reading}>
                        <Text style={styles.readingText}>
                            <TextBold>Value:</TextBold> {processReading(rowData.value, this.props.standard, 1) + " " + this.props.standard}{"\n"}
                            <TextBold>Date:</TextBold> {dateUtil.toDateTimeString(rowData.date)}
                        </Text>
                        <TouchableOpacity
                            style={styles.trashButton}
                            onPress={() => this.props.deleteReading(rowData)}>
                                <Icon style={{color:'white', fontSize:30}} theme={{iconFamily: "MaterialIcons"}} name="delete-forever"/>
                        </TouchableOpacity>
                    </View>
            }/>
        );
    }
}