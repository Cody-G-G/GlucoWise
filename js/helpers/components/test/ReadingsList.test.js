'use strict';
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import ReadingsList from "../ReadingsList";

test('ReadingsList - renders correctly', () => {
    const readings = [
        {id: '1', value: '50', date: new Date(2017, 2, 1, 20, 35, 10)},
        {id: '2', value: '60', date: new Date(2017, 2, 1, 20, 36, 5)},
        {id: '3', value: '70', date: new Date(2017, 2, 1, 20, 20, 10)}
    ];
    const standard = 'mg/dL';

    const tree = renderer.create(
        <ReadingsList readings={readings}
                      standard={standard}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});