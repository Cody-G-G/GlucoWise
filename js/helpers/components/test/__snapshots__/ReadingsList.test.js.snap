// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadingsList - renders correctly 1`] = `
<View
  style={undefined}
>
  <View>
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "royalblue",
                "color": "white",
                "flex": 6,
                "fontSize": 18,
                "padding": 7,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Value:
            </Text>
             
            50 mg/dL
            
            
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Date:
            </Text>
             
            01-03-2017  20:35
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "firebrick",
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
              theme={
                Object {
                  "iconFamily": "MaterialIcons",
                }
              }
            >
              юдл
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "cornflowerblue",
                "color": "white",
                "flex": 6,
                "fontSize": 18,
                "padding": 7,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Value:
            </Text>
             
            60 mg/dL
            
            
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Date:
            </Text>
             
            01-03-2017  20:36
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "firebrick",
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
              theme={
                Object {
                  "iconFamily": "MaterialIcons",
                }
              }
            >
              юдл
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "cornflowerblue",
                "color": "white",
                "flex": 6,
                "fontSize": 18,
                "padding": 7,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Value:
            </Text>
             
            70 mg/dL
            
            
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Date:
            </Text>
             
            01-03-2017  20:20
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "firebrick",
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
              theme={
                Object {
                  "iconFamily": "MaterialIcons",
                }
              }
            >
              юдл
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
